csharp/ql/test/query-tests/Security Features/CWE-079/XSSFlowAspNetCore/XSS.cs

```diff
@@ -0,0 +1,80 @@
+﻿
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Html;
+using Microsoft.Extensions.Primitives;
+
+namespace Testing.Controllers
+{
+    public class HomeViewModel
+    {
+        public string Message { get; set; }
+
+        public HtmlString Description { get; set; }
+    }
+
+    public class TestController : Controller
+    {
+        public IActionResult Index()
+        {
+            // BAD: flow of content type to.
+            var v = new ViewResult();
+            v.ViewData["BadData"] = new HtmlString(Request.Query["Bad data"]);
+
+            StringValues vOut;
+            Request.Query.TryGetValue("Foo", out vOut);
+
+            // BAD: via Enumerable.
```

Comment:
Does not appear in the expected output; either update query/library, or add `(false negative)` to the comment.